pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -352,3 +378,62 @@ func isMaster(machine mapiv1.Machine, client client.Client) bool {
 	}
 	return false
 }
+
+// checkAndDecrement checks if the provided MachineDisruptionBudget allows any disruption
+func (r *ReconcileMachineHealthCheck) checkAndDecrement(machineName string, mdb *healthcheckingv1alpha1.MachineDisruptionBudget, maxDisruptedMachinSize int) error {
+	if mdb.Status.ObservedGeneration < mdb.Generation {
```

Comment:
Suggestion, be pessimistic and respect the  .MachineDisruptionsAllowed < 0 and .DisruptedMachines > max conditions even if the values we are checking are old

Reply:
if this method returns an error, the MHC will skip fencing anyway